lib/season_stats.rb

```diff
@@ -0,0 +1,101 @@
+# separate it out to make sure it is only ONE season and be able to compare total wins for each team/coach witin a season
+
+class SeasonStats
+  attr_reader :games, :teams, :game_teams
+  def initialize(data)
+    @games = build_games(data)
+    @teams = build_teams(data)
+    @game_teams = build_game_teams(data)  
+  end
+
+  def build_games(data)
+    games = []
+    info = CSV.parse (File.read(data[:games])), headers: true, header_converters: :symbol
+    info.each do |row|
+      games << Game.new(row)
+    end
+    games
+  end
+
+  def build_teams(data)
+    teams = []
+    team_info = CSV.parse (File.read(data[:teams])), headers: true, header_converters: :symbol
+    team_info.each do |row|
+      teams << Team.new(row)
+    end
+    teams
+  end
+
+  
+  def build_game_teams(data)
+    game_teams = []
+    game_teams_info = CSV.parse (File.read(data[:game_teams])), headers: true, header_converters: :symbol
+    game_teams_info.each do |row|
+      game_teams << GameTeam.new(row)
+    end
+    game_teams
+  end
+  
+  def indiv_season(season_id)
+    season = games.find_all do |game|
```

Comment:
The .find_all enumerable works effectively. Why did you select that enumerable over each or others?

Reply:
hi! I chose the find_all so that it would pull ALL of the instances that matched my conditions in the code block below, since I wanted to have an array of all the games within the season!

Reply:
You gotta love that return value automatically pre-programmed in the enumerable.